### PR TITLE
additional properties could be false now

### PR DESCRIPTION
**Pull request by Arkady-Dymkov**
At the Json Scheme documentation additionalProperties could be false or map[string]*Type

### CODE REVIEW

File: pkg/generator/generate.go [hunk 1]
```diff
@@ -597,8 +597,9 @@ func (g *schemaGenerator) generateStructType(
 		}
 		valueType := codegen.Type(codegen.EmptyInterfaceType{})
 		var err error
-		if t.AdditionalProperties != nil {
-			if valueType, err = g.generateType(t.AdditionalProperties, nil); err != nil {
+		if b, ok := (*t.AdditionalProperties).(bool); t.AdditionalProperties != nil && (!b || ok) {
```
Comment (GreatG0ose) [inline pkg/generator/generate.go]: could you split this bool expression into separate blocks, so it is more readable?)

File: pkg/schemas/model.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ type Type struct {
 	Required             []string         `json:"required,omitempty"`             // section 5.15
 	Properties           map[string]*Type `json:"properties,omitempty"`           // section 5.16
 	PatternProperties    map[string]*Type `json:"patternProperties,omitempty"`    // section 5.17
-	AdditionalProperties *Type            `json:"additionalProperties,omitempty"` // section 5.18
+	AdditionalProperties *interface{}     `json:"additionalProperties,omitempty"` // section 5.18
```
Comment (GreatG0ose) [inline pkg/schemas/model.go:113]: not sure about using too generic type

File: pkg/generator/generate.go (hunk 1 above)
Comment (ViktorWii) [inline pkg/generator/generate.go:600]:
```suggestion
		b, ok := (*t.AdditionalProperties).(bool);
		if t.AdditionalProperties != nil && (!b || ok) {
```

Comment (VirtueOS) [reply pkg/generator/generate.go]:
```suggestion
		b, ok := (*t.AdditionalProperties).(bool);
		if t.AdditionalProperties != nil && (!b || ok) {
```

**Review by VirtueOS — APPROVED** [review]

File: pkg/schemas/model.go [hunk 3]
```diff
@@ -2,7 +2,7 @@
 //
 // Code borrowed from https://github.com/alecthomas/jsonschema/
 //
-// Copyright (C) 2014 Alec Thomas
+// # Copyright (C) 2014 Alec Thomas
```
Comment (VirtueOS) [inline pkg/schemas/model.go]: is this change intentional?

File: pkg/generator/generate.go [hunk 4]
```diff
@@ -597,8 +597,9 @@ func (g *schemaGenerator) generateStructType(
...
+			casted := (*t.AdditionalProperties).(*schemas.Type)
+			if valueType, err = g.generateType(casted, nil); err != nil {
```
Comment (VirtueOS) [inline pkg/generator/generate.go]: same goes here about splitting declaration and if statement
```suggestion
			valueType, err = g.generateType(casted, nil); 
			if err != nil {
```

File: pkg/schemas/model.go (hunk 2 above)
Comment (VirtueOS) [reply pkg/schemas/model.go:113]: @GreatG0ose 
Are you proposing to define new interface type for AdditionalProperties?

Comment (Arkady-Dymkov) [reply pkg/schemas/model.go:113]: I know nothing about "instance of" alternatives in go, but as far as I understand we could use *interface{}  or Any. Beat me if I'm wrong.

File: pkg/generator/generate.go (hunk 1 above)
Comment (VirtueOS) [reply pkg/generator/generate.go]: resolved in 380a1bd7176bcf472b3971531a8eeaa064f0ff14

File: pkg/generator/generate.go (hunk 4 above)
Comment (Arkady-Dymkov) [reply pkg/generator/generate.go]: Since additionalProperties in schema can be either false or Type, then we can pass casted value and use only casting status

File: pkg/schemas/model.go (hunk 2 above)
Comment (GreatG0ose) [reply pkg/schemas/model.go:113]: i propose nothing, that's just pointing at a risky piece of code here)

File: pkg/schemas/model.go (hunk 3 above)
Comment (GreatG0ose) [reply pkg/schemas/model.go]: does not seem resolved to me)

**Review by Arkady-Dymkov — COMMENTED** [review]
Best PR ever!

File: pkg/schemas/model.go (hunk 3 above)
Comment (VirtueOS) [reply pkg/schemas/model.go]:
```suggestion
// Copyright (C) 2014 Alec Thomas
```

File: pkg/generator/generate.go (hunk 4 above)
Comment (VirtueOS) [reply pkg/generator/generate.go]: how is this related to separation of declaration and if statement o.O

File: pkg/schemas/model.go (hunk 2 above)
Comment (Arkady-Dymkov) [reply pkg/schemas/model.go:113]: u r lead of this) Make ur decision)))

Comment (VirtueOS) [reply pkg/schemas/model.go:113]: postponed as the future problem
#2

**Review by GreatG0ose — APPROVED** [review]
Looks ok
